src/KeyHandlerTest.cpp

```diff
@@ -354,18 +355,19 @@ TEST_F(
   auto keys = asciiKeys("313");
   // ㄅˇ is not a viable composition.
   auto endState = handleKeySequence(keys, /*expectHandled=*/true,
-                                    /*expectErrorCallbackAtEnd=*/true);
-  auto emptyState = dynamic_cast<InputStates::Empty*>(endState.get());
-  ASSERT_TRUE(emptyState != nullptr);
+                                    /*expectErrorCallbackAtEnd=*/false);
+  auto inputtingState = dynamic_cast<InputStates::Inputting*>(endState.get());
+  ASSERT_TRUE(inputtingState != nullptr);
+  ASSERT_EQ(inputtingState->composingBuffer, "ˇ");
 }
 
 TEST_F(
     KeyHandlerTest,
     NonViableCompositionShouldRevertToEmptyStateIfComposingBufferEndsUpEmptyCase3) {
   auto keys = asciiKeys("31 ");
   // ㄅˇ is not a viable composition.
-  auto endState = handleKeySequence(keys, /*expectHandled=*/true,
-                                    /*expectErrorCallbackAtEnd=*/true);
+  auto endState = handleKeySequence(keys, /*expectHandled=*/false,
```

Comment:
Just `handleKeySequence(keys, /*expectedHadled=*/false)`, but probably we need a comment about why: first `31` resulting in the entire buffer being cleared, and so the last Space key just passes through?